### PR TITLE
Actually omit prowjobspec parts when empty

### PR DESCRIPTION
**Pull request by kargakis**
Fix the prowjob API to actually omit fields when empty.
Also fix some mutations in the plank controller that
will bite us back once we switch to informers w/ caches.

### CODE REVIEW

File: prow/cmd/deck/jobs.go [hunk 1]
```diff
@@ -207,10 +207,6 @@ func (ja *JobAgent) update() error {
 		buildID := j.Status.BuildID
 		nj := Job{
```
Comment (stevekuznetsov) [inline prow/cmd/deck/jobs.go]: I thought we were just going to get rid of this type?

Comment (kargakis) [reply prow/cmd/deck/jobs.go]: We will, there is an issue about it: https://github.com/kubernetes/test-infra/issues/5216

**Comment by stevekuznetsov** [comment]
OK, seems reasonable but hard to review the pointer diffs

/lgtm
/hold

**Pull request merged**